File: VB.NET/console/collection1/collection1/ienum1.vb
Imports System.Collections
Module ienum1
    Sub Main()
        Dim a As New ArrayList()
        a.Add(0)
        a.Add(1)
        a.Add(2)

        a.Add(3)
        a.Add(4)
        a.Add(5)
        Dim o As Object

        Console.WriteLine("Collection a: ")
        For Each o In a
            Console.Write(o & ", ")
        Next
        Dim i As IEnumerator

        i = a.GetEnumerator()
        Console.WriteLine("By Using Enumerator: ")
        While i.MoveNext()
            Console.WriteLine(i.Current)
        End While

        i = a.GetEnumerator(2, 3)
        Console.WriteLine("By Using Enumerator: ")
        While i.MoveNext()
            Console.WriteLine(i.Current)
        End While

    End Sub

End Module

File: csharp/console/rand1/rand1/ienum1.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace rand1
{
    class ienum1
    {
        static void Main(string[] ar)
        {
            ArrayList a= new ArrayList();
            a.Add("red");
            a.Add("blue");
            a.Add("green");
            a.Add("yellow");
            a.Add("beige");
            a.Add("brown");
            a.Add("magenta");
            a.Add("purple");

            IEnumerator e = a.GetEnumerator();
            while (e.MoveNext())
            {
                Object obj = e.Current;
                Console.WriteLine(obj);
            }

            Console.WriteLine();

            IEnumerator e2 = a.GetEnumerator(2,4);
            while (e2.MoveNext())
            {
                Object obj = e2.Current;
                Console.WriteLine(obj);
            }
   
        }
    }
}
